### PR TITLE
support actions on rows

### PR DESCRIPTION
**Pull request by iantrich**
Continuation of https://github.com/home-assistant/home-assistant-polymer/pull/3837 (Didn't want to deal with rebase when I'm changing the approach as well)

Docs: https://github.com/home-assistant/home-assistant.io/pull/10826

```yaml
type: entities
entities:
  - entity: sensor.breaking_changes
    hold_action:
      action: url
      url_path: https://www.home-assistant.io
```

### CODE REVIEW

File: src/panels/lovelace/cards/hui-entities-card.ts [hunk 1]
```diff
@@ -153,17 +155,36 @@ class HuiEntitiesCard extends LitElement implements LovelaceCard {
...
+    handleClick(this, this.hass!, config, false, false);
+  }
+
+  private _handleHold(ev: MouseEvent): void {
+    const config = (ev.currentTarget as any).config as any;
+    handleClick(this, this.hass!, config, true, false);
+  }
+
+  private _handleDblClick(ev: MouseEvent): void {
+    const config = (ev.currentTarget as any).entityConf as any;
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-entities-card.ts]:
```suggestion
    const config = (ev.currentTarget as any).config as any;
```

**Comment by bramkragten** [comment]
Have you convinced @balloob already about this?

**Comment by iantrich** [comment]
> Have you convinced @balloob already about this?

🙏

File: src/panels/lovelace/cards/hui-entities-card.ts (hunk 1 above)
Comment (iantrich) [reply src/panels/lovelace/cards/hui-entities-card.ts]: uh oh...I think I copied this from somewhere. Should double check the double tap PR, probably...

File: src/panels/lovelace/cards/hui-entities-card.ts [hunk 2]
```diff
@@ -153,17 +155,36 @@ class HuiEntitiesCard extends LitElement implements LovelaceCard {
       !DOMAINS_HIDE_MORE_INFO.includes(computeDomain(entityConf.entity))
```
Comment (iantrich) [inline src/panels/lovelace/cards/hui-entities-card.ts]: should probably update this to look at tap_event instead

File: src/panels/lovelace/cards/hui-entities-card.ts [hunk 3]
```diff
@@ -149,21 +151,41 @@ class HuiEntitiesCard extends LitElement implements LovelaceCard {
...
+        !DOMAINS_HIDE_MORE_INFO.includes(computeDomain(entityConf.entity)))
     ) {
       element.classList.add("state-card-dialog");
-      element.addEventListener("click", () => this._handleClick(entityConf));
     }
 
     return html`
-      <div>${element}</div>
+      <div
+        @ha-click=${this._handleClick}
```
Comment (balloob) [inline src/panels/lovelace/cards/hui-entities-card.ts]: Same issues like last time. We fight with the embedded element for clicks.

Comment (iantrich) [reply src/panels/lovelace/cards/hui-entities-card.ts]: But we already we're? In my testing I found no issues with interacting with the embedded elements

Comment (iantrich) [reply src/panels/lovelace/cards/hui-entities-card.ts]: @balloob is there anything else you want me to verify on this? I don't believe it is an issue.

**Comment by iantrich** [comment]
Moved the action handling to the generic-row's name after some discussion with @balloob.
Only issue I currently see is that the input-select row doesn't use the generic-row and only displays the state-badge to the left.

File: src/panels/lovelace/components/hui-generic-entity-row.ts [hunk 4]
```diff
@@ -132,6 +175,9 @@ class HuiGenericEntityRow extends LitElement {
         margin-left: 0;
         margin-right: 8px;
       }
+      .pointer {
+        cursor: pointer;
```
Comment (bramkragten) [inline src/panels/lovelace/components/hui-generic-entity-row.ts]: Let's add `padding: 12px 0px;` so the click target becomes a bit bigger

Comment (bramkragten) [reply src/panels/lovelace/components/hui-generic-entity-row.ts]: If there is no secondary that is... otherwise it is 4px padding I think?

**Comment by balloob** [comment]
I am fine with these changes if it does not hurt accessibility. Are we able to use the tab key to select the name and open the more info?

**Comment by iantrich** [comment]
> I am fine with these changes if it does not hurt accessibility. Are we able to use the tab key to select the name and open the more info?

We don't currently support this, the only thing that is selectable on the demo by tab are controls which is still true after my change.

**Comment by balloob** [comment]
Accessibility is very important. Not supporting that would literally cause us to exclude an already vulnerable part of the people. We've been slowly working on improving accessibility but still have a long way to go. Saying that we currently have a lot of big holes is not a reason to add to it. If we can't mark the element as interactive and allow tabbing to it to allow the user to activate the click handler using a keyboard, we shouldn't merge this. Double click and long press we can ignore, as they won't be part of the default generated UI config.

**Comment by iantrich** [comment]
@balloob I'm not taking away any accessibility as our current rows are not marked and I think it would be misleading to mark it as interactive now until we fix the click handler.

**Comment by balloob** [comment]
I'm down to merge this PR and add this feature, but we should make sure that it won't break our path to making these elements interactive. Because if that happens, we would end up having to remove it again, which pisses people off too. That's why I want to make sure that keyboard nav still works. And if we don't have examples of this working with long press directive, nor are we able to make it work, then the long press directive should not be used.

I think that it can work if we use a `<button>` or an `<a>` tag with `href` attribute + `preventDefault()` in the click listener.

**Comment by iantrich** [comment]
> That's why I want to make sure that keyboard nav still works

It doesn't work currently, is what I'm saying.

Regardless, I've pushed a commit that allows us to handle enter in long-press elements and made the state-badge (as input-select doesn't have a name in the row) to be tabbable and brings up the tap-action on enter.

File: src/panels/lovelace/common/directives/long-press-directive.ts [hunk 5]
```diff
@@ -138,9 +138,16 @@ class LongPress extends HTMLElement implements LongPress {
       window.setTimeout(() => (this.cooldownEnd = false), 100);
     };
 
+    const handleEnter = (ev: Event) => {
```
Comment (bramkragten) [inline src/panels/lovelace/common/directives/long-press-directive.ts]:
```
    const handleEnter = (ev: KeyboardEvent) => {
```

**Pull request merged**